From ee3c32917f22860dc1a0f08d2109dccd1466a3d8 Mon Sep 17 00:00:00 2001
From: Zumeng Chen <zumeng.chen@windriver.com>
Date: Thu, 28 Jul 2016 16:08:46 +0800
Subject: [PATCH 04/18] dpaa_eth: fix the wrong phandle to find ptp timer

The phandle of the node should be bit-exchanged before being used.

Signed-off-by: Zumeng Chen <zumeng.chen@windriver.com>
---
 drivers/net/ethernet/freescale/sdk_dpaa/dpaa_eth_common.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/net/ethernet/freescale/sdk_dpaa/dpaa_eth_common.c b/drivers/net/ethernet/freescale/sdk_dpaa/dpaa_eth_common.c
index fc6272c..0a101cd 100644
--- a/drivers/net/ethernet/freescale/sdk_dpaa/dpaa_eth_common.c
+++ b/drivers/net/ethernet/freescale/sdk_dpaa/dpaa_eth_common.c
@@ -595,7 +595,7 @@ dpa_mac_probe(struct platform_device *_of_dev)
 	if (phandle_prop && ((mac_dev->phy_if != PHY_INTERFACE_MODE_SGMII) ||
 			((mac_dev->phy_if == PHY_INTERFACE_MODE_SGMII) &&
 			 (mac_dev->speed == SPEED_1000)))) {
-		timer_node = of_find_node_by_phandle(*phandle_prop);
+		timer_node = of_find_node_by_phandle(be32_to_cpup(phandle_prop));
 		if (timer_node)
 			net_dev = dev_get_drvdata(dpa_dev);
 		if (timer_node && net_dev) {
-- 
2.7.4

